lib/Field/useLoader.js

```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react';
+
+export function useLoader(action, isInitiallyLoading = false) {
+  const [isLoading, setIsLoading] = useState(isInitiallyLoading);
+
+  const actionLoader = () => {
+    if (isLoading) {
+      return;
+    }
+    setIsLoading(true);
+    action(() => setIsLoading(false));
+  };
```

Comment:
🚩 I suspect that the most likely use case for this function will be with us passing in a promise as the action, so it's worth setting this up to accept a promise.

I think we will probably only use promises with this, so unless you have a specific non-promise use, I probably wouldn't bother with trying to make it handle both cases!

Pretty sure all you need to do is 
```suggestion
  const async actionLoader = () => {
    if (isLoading) {
      return;
    }
    setIsLoading(true);
    await action(() => setIsLoading(false));
  };
```

💚 In this case, we can also probably not bother with passing the callback to our action call, and just do
```suggestion
  const async actionLoader = () => {
    if (isLoading) {
      return;
    }
    setIsLoading(true);
    await action();
    setIsLoading(false)
  };
```
this is more of an opinionated comment, but I think if I was picking this component up I would expect it to stop loading once the promise passed was resolved. We also don't have to manually set is loading to false.

Reply:
Great suggestion, thanks!